test/extensions/filters/http/buffer/buffer_filter_integration_test.cc

```diff
@@ -33,4 +33,88 @@ TEST_P(BufferIntegrationTest, RouterRequestAndResponseWithZeroByteBodyBuffer) {
   testRouterRequestAndResponseWithBody(0, 0, false);
 }
 
+TEST_P(BufferIntegrationTest, RouterRequestBufferLimitExceeded) {
```

Comment:
One thought; when the per-route override is just the core config and you merge the two, I think we could do testing and the merging in some shared library between filters. The key advantage here is that we make it fairly trivial for filters to add a per-route override without having to write all this domain specific test code.